packages/phishing-controller/src/PhishingController.ts

```diff
@@ -498,6 +527,105 @@ export class PhishingController extends BaseController<
     );
   }
 
+  /**
+   * Checks if a patch represents a transaction-level change or nested transaction property change
+   *
+   * @param patch - Immer patch to check
+   * @returns True if patch affects a transaction or its nested properties
+   */
+  #isTransactionPatch(patch: Patch): boolean {
+    const { path } = patch;
+    return (
+      path.length >= 2 &&
+      path[0] === 'transactions' &&
+      typeof path[1] === 'number'
+    );
+  }
+
+  /**
+   * Handle transaction controller state changes using Immer patches
+   * Extracts token addresses from simulation data and groups them by chain for bulk scanning
+   *
+   * @param _state - The current transaction controller state
+   * @param _state.transactions - Array of transaction metadata
+   * @param patches - Array of Immer patches only for transaction-level changes
+   */
+  #onTransactionControllerStateChange(
+    _state: { transactions: TransactionMeta[] },
+    patches: Patch[],
+  ) {
+    try {
+      const tokensByChain = new Map<string, Set<string>>();
+
+      for (const patch of patches) {
+        if (patch.op === 'remove') {
+          continue;
+        }
+
+        // Handle transaction-level patches (includes simulation data updates)
+        if (this.#isTransactionPatch(patch) && patch.path.length === 2) {
```

Comment:
We can remove the `path.length === 2` condition, and add it to the isTransactionPatch (which is already validating `path.length >= 2`)